### PR TITLE
chore: refactor to modular cli

### PR DESCRIPTION
**Pull request by yehjxraymond**
See: https://github.com/yargs/yargs/blob/master/docs/advanced.md#commanddirdirectory-opts

### CODE REVIEW

**Comment by Nebulis** [comment]
I'm definitely not fond of the separation between commands and implementations folder, would rather have everything together :/

File: src/__tests__/errors.test.ts [hunk 1]
```diff
@@ -1,149 +0,0 @@
-import { transformAdditionalPropertyErrors, transformAllowedValuesErrors, transformRequiredErrors } from "../errors";
```
Comment (Nebulis) [inline src/__tests__/errors.test.ts]: why removed ?

Comment (yehjxraymond) [reply src/__tests__/errors.test.ts]: yargs handles everything now

File: src/errors.ts [hunk 2]
```diff
@@ -1,72 +0,0 @@
-import Ajv, { AdditionalPropertiesParams, EnumParams, RequiredParams } from "ajv";
```
Comment (Nebulis) [inline src/errors.ts]: why remove this ?

File: src/__tests__/errors.test.ts (hunk 1 above)
Comment (Nebulis) [reply src/__tests__/errors.test.ts]: this test errors from ajv

File: src/errors.ts (hunk 2 above)
Comment (yehjxraymond) [reply src/errors.ts]: no longer used as we are not validating the passed in commands now. It's not used in any of the other implementations as well

File: src/__tests__/errors.test.ts (hunk 1 above)
Comment (yehjxraymond) [reply src/__tests__/errors.test.ts]: moved it within wrap

File: src/errors.ts (hunk 2 above)
Comment (yehjxraymond) [reply src/errors.ts]: moved it within wrap

**Pull request merged**